### PR TITLE
syncer: fix syncer not being triggered by ticker

### LINKED ISSUE #1798: secondsBehindMaster didn't return 0 after workload is gone
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

https://asktug.com/t/topic/93842

2. What did you expect to see?

3. What did you see instead?

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 
close #1798 

### What is changed and how it works?

as title

### Check List 

Tests 

 - Integration test

Code changes

Side effects
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -1192,6 +1192,7 @@ func (s *Syncer) syncDML(tctx *tcontext.Context, queueBucket string, db *DBConn,
 
 	var err error
 	var affect int
+	ticker := time.NewTicker(waitTime)
```
Comment (lance6716) [inline syncer/syncer.go:1195]: ticker.Stop()

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lichunzhu** [comment]
Is this bug caused by `successF` isn't triggered every time?
If so, I don't think this PR solved this bug.
@Ehco1996 @lance6716

**Comment by GMHDBJD** [comment]
> Is this bug caused by `successF` isn't triggered every time?
> If so, I don't think this PR solved this bug.
> @Ehco1996 @lance6716

You are right. 🤣

**Comment by Ehco1996** [comment]
> Is this bug caused by `successF` isn't triggered every time?
> If so, I don't think this PR solved this bug.
> @Ehco1996 @lance6716

@lichunzhu  i don't understand this, `successF` only called after `executeSQLs` success . and 

the root cause is when no job in queue, lag will never be updated because ticker not work

**Comment by lichunzhu** [comment]
@Ehco1996 
After this PR, lag update logic will still not be used. Change time after to ticker doesn't work. Because job length is still zero.

**Comment by Ehco1996** [comment]
> @Ehco1996
> After this PR, lag update logic will still not be used. Change time after to ticker doesn't work. Because job length is still zero.

oh...i know,i will provide another PR to fix this

**Comment by lance6716** [comment]
this PR is wrong because ticker will be generating tick at background, however we require wait 100ms after entering `select`. 

please help us fix it in https://github.com/pingcap/dm/pull/1855 @Ehco1996

**Comment by lance6716** [comment]
maybe we should call `Ticker.Reset` @Ehco1996

**Comment by glorv** [comment]
> maybe we should call `Ticker.Reset` @Ehco1996

In this way, we can simply call `time.After` instead

**Comment by lance6716** [comment]
> > maybe we should call `Ticker.Reset` @Ehco1996
> 
> In this way, we can simply call `time.After` instead

OK. I'm a little bit wory about performance when concurrency is high 
image

**Comment by glorv** [comment]
ok. Then maybe we should call `timer.Reset` at each loop start. 😂️

**Pull request merged**